libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveBotComponent.cs

```diff
@@ -93,7 +93,7 @@ public override void ConfigureServices(IServiceCollection services, IConfigurati
             services.AddSingleton<DeclarativeType>(sp => new DeclarativeType<EmitEvent>(EmitEvent.Kind));
             services.AddSingleton<DeclarativeType>(sp => new DeclarativeType<EndDialog>(EndDialog.Kind));
             services.AddSingleton<DeclarativeType>(sp => new DeclarativeType<EndTurn>(EndTurn.Kind));
-            services.AddSingleton<DeclarativeType>(sp => new DeclarativeType<Foreach>(Foreach.Kind));
+            services.AddSingleton<DeclarativeType>(sp => new DeclarativeType<ForEachElement>(ForEachElement.Kind));
```

Comment:
Are we breaking backward compat?

Reply:
ForEach could not be fixed without having new methods, so this is why I wrote a new one and marked the old one as obsolete. For composer users, no change should be experienced (except that foreach will now work properly). And if someone derived from ForEach, they can still use the subclass, which is why I left ForEAch as a public class instead of refactoring.